vars/publishIntegTestResults.groovy

```diff
@@ -0,0 +1,226 @@
+                            },
+                            "with_security_cluster_stdout": {
+                                "type": "keyword"
+                            },
+                            "with_security_cluster_stderr": {
+                                "type": "keyword"
+                            },
+                            "without_security": {
+                                "type": "keyword"
+                            },
+                            "without_security_build_yml": {
+                                "type": "keyword"
+                            },
+                            "without_security_cluster_stdout": {
+                                "type": "keyword"
+                            },
+                            "without_security_cluster_stderr": {
+                                "type": "keyword"
+                            }
+                        }
+                    }
+                }'
+                curl -I "${METRICS_HOST_URL}/${indexName}" --aws-sigv4 \"aws:amz:us-east-1:es\" --user \"${awsAccessKey}:${awsSecretKey}\" -H \"x-amz-security-token:${awsSessionToken}\" | grep -E "HTTP\\/[0-9]+(\\.[0-9]+)? 200"
+                if [ \$? -eq 0 ]; then
+                    echo "Index already exists. Indexing Results"
+                else
+                    echo "Index does not exist. Creating..."
+                    create_index_response=\$(curl -s -XPUT "${METRICS_HOST_URL}/${indexName}" --aws-sigv4 \"aws:amz:us-east-1:es\" --user \"${awsAccessKey}:${awsSecretKey}\" -H \"x-amz-security-token:${awsSessionToken}\" -H 'Content-Type: application/json' -d "\${INDEX_MAPPING}")
+                    if [[ \$create_index_response == *'"acknowledged":true'* ]]; then
+                        echo "Index created successfully."
+                    else
+                        echo "Failed to create index. Error message: \$create_index_response"
+                        exit 1
+                    fi
+                fi
+                if [ -s ${testRecordsFile} ]; then
+                    echo "File Exists, indexing results."
+                    curl -XPOST "${METRICS_HOST_URL}/$indexName/_bulk" --aws-sigv4 \"aws:amz:us-east-1:es\" --user \"${awsAccessKey}:${awsSecretKey}\" -H \"x-amz-security-token:${awsSessionToken}\" -H "Content-Type: application/x-ndjson" --data-binary "@${testRecordsFile}"
+                else
+                    echo "File Does not exist. No tests records to process."
+                fi
+        """
+        }
+    }
+}
+
+def generateJson(component, version, integTestBuildNumber, integTestBuildUrl, distributionBuildNumber, distributionBuildUrl, buildStartTime, rc, rcNumber, platform, architecture, distribution, componentCategory, componentResult, testReportManifestYmlUrl, withSecurity, withSecurityYml, withSecurityStdout, withSecurityStderr, withoutSecurity, withoutSecurityYml, withoutSecurityStdout, withoutSecurityStderr) {
```

Comment:
Possible to combine both generateJson block to a common lib?
Or maybe rename to a different func name to avoid confusion?

Thanks.

Reply:
Thank you. The `generateJson` method generates different `key:value` paired documents for each use case, such as for distribution builds and integration tests. Therefore, combining them into a common method would be overkill, as the purpose of `generateJson` is unique for each use case. As for the naming, I kept `generateJson` constant for both `publishDistributionBuildResults` and `publishIntegTestResults` because the purpose is the same and it is easily understood by developers. The `generateJson` should be an interface method but not possible with groovy jenkins libraries, here is some discussion from past https://github.com/opensearch-project/opensearch-build-libraries/pull/436#discussion_r1637169462.

Reply:
Got it, thanks.